ui/app/AppLayouts/Profile/helpers/ProfileShowcaseModels.qml

```diff
@@ -0,0 +1,163 @@
+    }
+
+    function changeCommunityPosition(key, to) {
+        communities.changePosition(key, to)
+    }
+
+    // ACCOUNTS
+
+    // Input models
+    property alias accountsSourceModel: accounts.sourceModel
+    property alias accountsShowcaseModel: accounts.showcaseModel
+
+    // Output models
+    readonly property alias accountsVisibleModel: accounts.visibleModel
+    readonly property alias accountsHiddenModel: accounts.hiddenModel
+
+    // Methods
+    function accountsCurrentState() {
+        return accounts.currentState()
+    }
+
+    function setAccountVisibility(key, visibility) {
+        accounts.setVisibility(key, visibility)
+    }
+
+    function changeAccountPosition(key, to) {
+        accounts.changePosition(key, to)
+    }
+
+    // Other
+    readonly property alias visibleAccountsList:
+        visibleAccountsConnections.visibleAccountsList
+
+    readonly property alias accountsVisibilityMap:
+        visibleAccountsConnections.accountsVisibilityMap
+
+    // COLLECTIBLES
+
+    // Input models
+    property alias collectiblesSourceModel: collectiblesFilter.sourceModel
+    property alias collectiblesShowcaseModel: collectibles.showcaseModel
+
+    // Output models
+    readonly property alias collectiblesVisibleModel: collectibles.visibleModel
+    readonly property alias collectiblesHiddenModel: collectibles.hiddenModel
+
+    // Methods
+    function collectiblesCurrentState() {
+        return collectibles.currentState()
+    }
+
+    function setCollectibleVisibility(key, visibility) {
+        collectibles.setVisibility(key, visibility)
+    }
+
+    function changeCollectiblePosition(key, to) {
+        collectibles.changePosition(key, to)
+    }
+
+    ProfileShowcaseDirtyState {
+        id: communities
+    }
+
+    ProfileShowcaseDirtyState {
+        id: accounts
+    }
+
+    ProfileShowcaseDirtyState {
+        id: collectibles
+
+        sourceModel: collectiblesFilter
+    }
+
+    SortFilterProxyModel {
+        id: collectiblesFilter
+
+        delayed: true
+
+        proxyRoles: FastExpressionRole {
+            name: "maxVisibility"
+
+            expression: {
+                const m = root.accountsVisibilityMap
+                const accounts = model.accounts.split(":")
+                const visibilities = accounts.map(e => m[e]).filter(e => e)
+
+                return visibilities.length ? Math.min(...visibilities) : 0
+            }
+
+            expectedRoles: ["accounts"]
+        }
+
+        filters: RangeFilter {
+            roleName: "maxVisibility"
+            minimumValue: 1
+        }
+    }
+
+    Connections {
+        id: visibleAccountsConnections
+
+        target: accounts.visibleModel
+
+        property var visibleAccountsList: []
+        property var accountsVisibilityMap: ({})
+
+        function updateAccountsList() {
+            const keysAndVisibility = ModelUtils.modelToArray(
+                        accounts.visibleModel, ["key", "visibility"])
```

Comment:
For each profile showcase section key role name is different, is it possible to configure it for helper models?

Reply:
So let's see how it will look like finally. There are several options to adjust that. By parametrization here or e.g. using `RenamingModel`.